packages/nextjs/app/builders/_components/MemberList.tsx

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useState } from "react";
+import Link from "next/link";
+import { Address } from "~~/components/scaffold-eth";
+import { useScaffoldEventHistory } from "~~/hooks/scaffold-eth";
+
+interface ValidPages {
+  [key: string]: boolean;
+}
+
+export const MemberList = () => {
+  const {
+    data: events = [],
+    isLoading: isLoadingEvents,
+    error: errorReadingEvents,
+  } = useScaffoldEventHistory({
+    contractName: "BatchRegistry",
+    eventName: "CheckedIn",
+    fromBlock: 122221841n,
+    watch: true,
+  });
+
+  const [validPages, setValidPages] = useState<ValidPages>({});
+
+  useEffect(() => {
+    const checkPages = async () => {
+      const results: ValidPages = {};
+      for (const event of events) {
+        const builderAddress = event.args.builder;
+        if (builderAddress) {
+          const response = await fetch(`/builders/${builderAddress}`);
+          results[builderAddress] = response.ok;
+        }
+      }
+      setValidPages(results);
+    };
+
+    if (events.length > 0) {
+      checkPages();
+    }
+  }, [events]);
+
+  if (isLoadingEvents)
+    return (
+      <div className="flex justify-center items-center h-full">
+        <p>Loading events...</p>
+      </div>
+    );
+  if (errorReadingEvents)
+    return (
+      <div className="flex justify-center items-center h-full">
+        <p>Error loading events: {errorReadingEvents.message}</p>
+      </div>
+    );
+
+  return (
+    <div className="container mx-auto p-4">
+      <table className="table-auto w-full text-left border-collapse">
+        <thead>
+          <tr className="bg-secondary">
+            <th className="px-4 py-2 w-1/2 border-b-2">Builder</th>
+            <th className="px-4 py-2 w-1/2 border-b-2">Builder Page</th>
+          </tr>
+        </thead>
+        <tbody>
+          {events.map(event => {
```

Comment:
Since in contract same builder can checkIn multiple time we should make sure that we only display first time checksIn, we could  probably do `event.filter(event => event.args.first).map(...)`